assertk/src/commonMain/kotlin/assertk/assertions/throwable.kt

```diff
@@ -62,5 +62,16 @@ fun Assert<Throwable>.hasRootCause(cause: Throwable) {
     }
 }
 
+/**
+ * Asserts the throwable with a specific type have the expected properties for it.
+ */
+fun <T: Throwable> Assert<T>.hasProperties(vararg pairs: Pair<KProperty1<T, Any>, Any>) {
```

Comment:
This is unrelated to `assertFailsWith`. It's not clear why it's type parameter is bound to `Throwable` subtypes since it's a general-purpose utility. I would separate it into its own PR, or even start with an issue discussing the use case and why regular `.all { }` in insufficient.